### ISSUE TITLE
onNotificationOpen not called on ios 12

### ISSUE THREAD

**Issue by echonox** [arnesson/cordova-plugin-firebase#912]
**Describe the bug**
I receive push notification when in background, but onNotificationOpen doesn't get called in foreground and neither background

When receiving the notification I get:

[Firebase/Messaging][I-FCM002019] FIRMessaging received data-message, but FIRMessagingDelegate's-messaging:didReceiveMessage: not implemented

**To Reproduce**
1. Install Xcode 10
2. create new cordova project
3. add the latest stable firebase plugin
4. set all config
5. send push notification

**Expected behavior**
I expect onNotificationOpen to execute and to see the notification data on js side

**Plugin Version**
2.0.5

**Desktop (please complete the following information):**
 - OS: OS X High Sierra
 - Browser safari
 - Version 12.0

**Smartphone (please complete the following information):**
 - Device: iPhone 5S
 - OS: iOS 12
 - Browser safari

**Additional context**
NOTE: push notification and background modes are enabled under capabilities!

**Comment by briantq** [comment]
I'm on my mobile phone so can't find the exact issue at the moment, but believe you are encountering the following Cordova bug https://github.com/apache/cordova-ios/issues/407. Please see if that resolved your issue.

**Comment by echonox** [comment]
@briantq yes I build the app with the legacy build, I've found many issues about this and with a PR that it got fixed but i still receive:

[Firebase/Messaging][I-FCM002019] FIRMessaging received data-message, but FIRMessagingDelegate's-messaging:didReceiveMessage: not implemented

**Comment by kamaleshbasu** [comment]
@echonox Did you manage to find a solution to this? I am still experiencing this with iOs 12. It works perfectly when app is in background but doesn't when app is in foreground.

Android works perfectly.
